docs/architecture/api-20201231-preview.md

```diff
@@ -0,0 +1,148 @@
+# API 20201231-preview design
+
+For api `20201231-preview` bellow is the list of proposed changes and features.
+These would have to land into API and backend as a features.
+
+Feature work will need to be agreed and converted to working stories. All technical details bellow are overview only
+
+## API changes
+
+### ComputeSecurityProfile
+
+Add security profile to worker and master pools configuration for all security enhancements
+for the compute instances
+
+We need to make sure that these option are validated on the cluster too when customer is interacting with MachineSet objects. We should be able to verify if these options where enabled on cluster create and set those accordingly. This does not prevent customer to create instances with and without encryption in the same cluster.
+
+```
+// MasterProfile represents a master profile.
+type MasterProfile struct {
+	...
+
+	ComputeSecurityProfile
+}
+
+
+// WorkerProfile represents a worker profile.
+type WorkerProfile struct {
+	...
+
+	ComputeSecurityProfile
+}
+```
+
+// EncryptionAtHostEnum enumerates the values for Encryption at host
+type EncryptionAtHostEnum string
+
+const (
+	// Disabled ...
+	Disabled EncryptionAtHostEnum = "Disabled"
```

Comment:
EncryptionAtHostDisabled EncryptionAtHost = "Disabled"

Reply:
This matches azure SDK, but I dont mind either way.